files/galaxy-test/tpv/tool_defaults.yml

```diff
@@ -25,6 +25,17 @@ tools:
           training_labels = '"' + ", ".join(training_roles) + '"'
           entity.params['requirements'] = '(GalaxyGroup == "compute") || (%s)' % training_expr if training_expr else '(GalaxyGroup == "compute")'
           entity.params['+Group'] = training_labels
+      - id: remote_resources
+        if: user is not None
+        execute: |
+          user_preferences = user.extra_preferences
+          for data_item in user_preferences:
+            if "distributed_compute|remote_resources" in data_item:
+                if user_preferences[data_item] != "None":
+                    pulsar_tag = user_preferences[data_item]
+          new_tag = [{ 'require': pulsar_tag }] if pulsar_tag else []
```

Comment:
This may error out with `NameError: name 'pulsar_tag' is not defined` whenever `"distributed_compute|remote_resources" not in data_item or user_preferences[data_item] != "None".`

Reply:
Thanks for changing it.